fetching/projio.py

```diff
@@ -8,11 +10,19 @@ class Project:
     """
 
     def __init__(self, root: pathlib.Path):
+        assert root.is_dir()
         self.root = root
 
-    @property
+    @functools.cached_property
     def test_directory(self) -> pathlib.Path | None:
```

Comment:
Add is_ .

Reply:
Why? this method does not return boolean. This is an Optional